src/brogue/Config.c

```diff
@@ -0,0 +1,255 @@
+static const char* variantMappings[] = {"brogue", "rapid", NULL};
+
+// maps json strings to the graphicsModes Enum
+static const char* graphicsModeMappings[] = {"text", "tiles", "hybrid", NULL};
+
+static intRange playbackDelayRange = {MIN_PLAYBACK_DELAY, MAX_PLAYBACK_DELAY};
+
+static configParams createDefaultConfig() {
+    configParams config;
+
+    config.playbackDelayPerTurn = DEFAULT_PLAYBACK_DELAY;
+    config.gameVariant = VARIANT_BROGUE;
+    config.graphicsMode = TEXT_GRAPHICS;
+    config.displayStealthRangeMode = false;
+    config.trueColorMode = false;
+    config.easyMode = false;
+    config.wizard = false;
+
+    return config;
+}
+
+static char* loadConfigFile() {
+    FILE* jsonFile = fopen(JSON_FILENAME, "r");
+
+    if (!jsonFile) {
+        return NULL;
+    }
+
+    fseek(jsonFile, 0, SEEK_END);
+    long fileSize = ftell(jsonFile);
+    fseek(jsonFile, 0, SEEK_SET);
+
+    char* buffer = (char*)malloc(fileSize + 1);
+
+    if (!buffer) {
+        fclose(jsonFile);
+        return NULL;
+    }
+
+    fread(buffer, 1, fileSize, jsonFile);
+    fclose(jsonFile);
+
+    buffer[fileSize] = '\0';
+
+    return buffer;
+}
+
+static configField* getFieldEntries(configParams* config) {
+    int numFields = 7;
+
+    configField fieldDescriptors[] = {
+        {"gameVariant", &(config->gameVariant), ENUM_STRING, {.enumMappings = variantMappings}},
+        {"graphicsMode", &(config->graphicsMode), ENUM_STRING, {.enumMappings = graphicsModeMappings}},
+        {"playbackDelayPerTurn", &(config->playbackDelayPerTurn), INT_TYPE, {.intRange = playbackDelayRange}},
+        {"displayStealthRangeMode", &(config->displayStealthRangeMode), BOOLEAN_TYPE},
+        {"trueColorMode", &(config->trueColorMode), BOOLEAN_TYPE},
+        {"wizard", &(config->wizard), BOOLEAN_TYPE},
+        {"easyMode", &(config->easyMode), BOOLEAN_TYPE},
+        {NULL}};
+
+    configField* entries = calloc(numFields + 1, sizeof(configField));
+
+    for (int i = 0; i < numFields; i++) {
+        entries[i] = fieldDescriptors[i];
+    }
+
+    return entries;
+}
+
+static short mapStringToEnum(const char* inputString, const char** mappings) {
+    for (short i = 0; mappings[i]; i++) {
+        if (strcmp(inputString, mappings[i]) == 0) {
+            return i;
+        }
+    }
+    return -1;
+}
+
+static void parseConfigValues(const char* jsonString, configParams* config) {
+    if (!jsonString || !config) {
+        return; // Invalid input
+    }
+
+    cJSON* root = cJSON_Parse(jsonString);
+
+    if (!root) {
+        return; // JSON parsing error
+    }
+
+    configField* entries = getFieldEntries(config);
+
+    for (int i = 0; entries[i].name; i++) {
+        cJSON* jsonField = cJSON_GetObjectItem(root, entries[i].name);
+
+        if (jsonField) {
+            switch (entries[i].type) {
+            case INT_TYPE:
+                if (cJSON_IsNumber(jsonField)) {
+                    short value = jsonField->valueint;
```

Comment:
Need to check for overflow/underflow, since `jsonField->valueint` is an `int` (typically 32 bits) but `value` is a `short` (typically 16 bits).